src/sentry/api/endpoints/group_events.py

```diff
@@ -101,4 +104,11 @@ def respond(queryset):
 
             events = events.filter(id__in=event_ids)
 
+        # filter out events which are beyond the retention period
+        retention = quotas.get_event_retention(organization=group.project.organization)
+        if retention:
+            events = events.filter(
+                datetime__gte=timezone.now() - timedelta(days=retention)
+            )
```

Comment:
Any reason not to make this a helper on the manager or something so it's not repeated?

Reply:
@bretthoerner probably worth while -- the problem is managers kind of suck in django because you have to like

Foo.objects.with_retention.blbhalbha

you cant Foo.objects.filter().with_retent().blahblah

Reply:
> you cant Foo.objects.filter().with_retent().blahblah

You can if you add it to the `QuerySet` instead of the `Manager` fwiw.

The manager has a `get_queryset` method, so you can use that to return a custom `QuerySet` class which has whatever methods you want on it.

Reply:
I realize that

Reply:
In that case, you'd have to have the method on both the manager and the `QuerySet` though, but the manager just looks like:

```python
def with_retent(self, *args, **kwargs):
    return self.get_queryset().with_retent(*args, **kwargs)
```

:)